clang/lib/Frontend/InitPreprocessor.cpp

```diff
@@ -926,6 +926,12 @@ static void InitializePredefinedMacros(const TargetInfo &TI,
   if (LangOpts.GNUCVersion && LangOpts.CPlusPlus11)
     Builder.defineMacro("__GXX_EXPERIMENTAL_CXX0X__");
 
+  if (TI.getTriple().isWindowsGNUEnvironment() && LangOpts.CPlusPlus) {
```

Comment:
The CPlusPlus condition should probably be removed.

```
% x86_64-w64-mingw32-gcc -dM -xc -E /dev/null G TYPEINFO
#define __GXX_TYPEINFO_EQUALITY_INLINE 0
#define __GXX_MERGED_TYPEINFO_NAMES 0
```

Reply:
Ok, done. Yeah while it doesn't serve much purpose in non-c++ mode, it doesn't hurt either and simplifies things a little bit.